### PR TITLE
Implement notunder(...) syntax

### PR DESCRIPTION
**Pull request by AlexanderKapitanovski**
added notunder() syntax (multi node choose possible) and multi node choose to under()

### CODE REVIEW

File: src/StructuredLogViewer/Controls/BuildControl.xaml.cs [hunk 1]
```diff
@@ -914,6 +919,15 @@ public void SearchInSubtree()
             }
         }
 
+        public void ExcludeSubtreeFromSearch()
+        {
+            var treeNode = treeView.SelectedItem as TimedNode;
+            if (treeNode != null)
```
Comment (KirillOsenkov) [inline src/StructuredLogViewer/Controls/BuildControl.xaml.cs:925]: `if (treeView.SelectedItem is TimedNode treeNode)`

File: src/StructuredLogViewer.Core/NodeQueryMatcher.cs [hunk 2]
```diff
@@ -69,9 +56,23 @@ public NodeQueryMatcher(string query, IEnumerable<string> stringTable)
...
                 {
-                    word = word.Substring(6, word.Length - 7);
+                    var nodeIndexText = word.Substring(7, word.Length - 8);
                     Words.RemoveAt(i);
-                    UnderMatcher = new NodeQueryMatcher(word, stringTable);
+                    if (int.TryParse(nodeIndexText, out var nodeIndex))
+                    {
+                        UnderNodes.Add(nodeIndex);
+                    }
+                    continue;
```
Comment (KirillOsenkov) [inline src/StructuredLogViewer.Core/NodeQueryMatcher.cs:65]: we still need to have an UnderMatcher if the query is not a node index.

You can have queries like `$task Csc under($project Foo)`, so UnderMatcher recursively evaluates the expression in parentheses `$project Foo`

File: src/StructuredLogViewer.Core/NodeQueryMatcher.cs [hunk 3]
```diff
@@ -22,6 +21,8 @@ public class NodeQueryMatcher
         private bool searchInValue { get; set; }
         private string nameToSearch { get; set; }
         private string valueToSearch { get; set; }
+        private HashSet<int> UnderNodes { get; set; } = new HashSet<int>();
```
Comment (KirillOsenkov) [inline src/StructuredLogViewer.Core/NodeQueryMatcher.cs:24]: I think these will need to be converted to `List<NodeQueryMatcher>` to allow for arbitrary expressions in parentheses

Comment (AlexanderKapitanovski) [reply src/StructuredLogViewer.Core/NodeQueryMatcher.cs:24]: sorry I didn't see this tipp $task Csc under($project Foo) and didn't imagine this way to search.  I will make it better.:-)

Comment (AlexanderKapitanovski) [reply src/StructuredLogViewer.Core/NodeQueryMatcher.cs]: I've changed the logik to allow arbitrary expression in under() notunder(), need to look at this together

**Pull request merged**